src/NServiceBus.RabbitMQ/Configuration/ConnectionStringParser.cs

```diff
@@ -4,42 +4,47 @@
     using System.ComponentModel;
     using System.Data.Common;
     using System.Linq;
-    using System.Reflection;
     using Logging;
     using Settings;
 
     class ConnectionStringParser : DbConnectionStringBuilder
     {
         static readonly ILog Logger = LogManager.GetLogger(typeof(ConnectionStringParser));
 
-        readonly ReadOnlySettings settings;
+        readonly SettingsHolder settings;
 
-        public ConnectionStringParser(ReadOnlySettings settings)
+        public ConnectionStringParser(SettingsHolder settings)
         {
             this.settings = settings;
         }
 
-        public ConnectionConfiguration Parse(string connectionString)
+        public void Parse(string connectionString)
         {
             ConnectionString = connectionString;
 
-            var connectionConfiguration = new ConnectionConfiguration(settings);
-            var connectionConfigurationType = typeof(ConnectionConfiguration);
+            DefaultConfigurationValues.Apply(settings);
 
             foreach (var key in Keys.Cast<string>())
             {
-                var property = connectionConfigurationType.GetProperty(key, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance);
-                property?.SetValue(connectionConfiguration, TypeDescriptor.GetConverter(property.PropertyType).ConvertFrom(this[key]));
+                var normalizedKey = string.Format("RabbitMQ.{0}", key);
+                if (settings.HasSetting(normalizedKey))
+                {
+                    settings.Set(normalizedKey, TypeDescriptor.GetConverter(settings.Get(normalizedKey).GetType()).ConvertFrom(this[key]));
```

Comment:
THis is a line that I don't particularly like. At the moment one of our setting (CertPassphrase) has default value null => this line was throwing null ref expcetion. For the time being I changed default value to `string.Empty` only to not spend to much time how to solve this piece of a puzzle.